Update charts for 1.9 rebase

shamsher31 opened the pull request:
Note that as a part of the 1.9 rebase, the `global.yaml` is removed but we will have to keep it in order to avoid breaking changes. Also `Mesh expansion` is removed from the charts. Let me know if I miss something.

Commit:
Update chart for 1.9 rebase

Commit:
updated chart

shamsher31 commented:
/test istio-operator_unittests

shamsher31 commented on resources/helm/v2.1/gateways/istio-egress/templates/injected-deployment.yaml:
```diff
@@ -0,0 +1,141 @@
+    spec:
+{{- if not $gateway.runAsRoot }}
+      securityContext:
+        runAsUser: 1337
```
I think this is the new file `injected-deployment.yaml` added in 1.9 rebase. Do we need to remove this permission? like we are doing here https://github.com/maistra/istio-operator/blob/maistra-2.1/build/patch-charts.sh#L276

rcernich replied:
Use the other gateway deployment template as an example.  For those, it looks like we set securityContext.runAsNonRoot=true in this case.

rcernich replied:
Compare this whole file with the standard gateway deployment template.  They should be fairly similar.  Note that this is the deployment used with injection, so much of the istio-proxy container configuration will be applied when istiod injects it into the pod.

shamsher31 replied:
Updated [here](https://github.com/maistra/istio-operator/pull/737/commits/95d20365e7ccbf08d878b05a49649d5fa9e565c7). Thanks

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/deployment.yaml:
```diff
@@ -59,12 +68,14 @@
-  name: {{ $gateway.name | default "istio-egressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be updated to match the old version or it will break installing gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/service.yaml:
```diff
@@ -14,14 +15,32 @@ metadata:
 spec:
-  type: ClusterIP
+{{- if $gateway.loadBalancerIP }}
+  loadBalancerIP: "{{ $gateway.loadBalancerIP }}"
```
We'll need to verify whether or not these are supported by SMCP and if not, do we need to support them.

shamsher31 replied:
@luksa can you please provide your inputs here?

luksa replied:
Looks like we do support this in the SMCP API: https://github.com/Maistra/istio-operator/blob/8ababc86c5da3336f5471ee9fbf72240d5c6e13b/pkg/apis/maistra/v2/gateways.go#L53

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/service.yaml:
```diff
@@ -14,14 +15,32 @@ metadata:
       {{- end }}
     {{- end }}
+
+  {{ range $app := $gateway.egressPorts }}
```
We need to determine if we need to support this new setting `egressPorts`.  I don't know why this would be needed, as we already have `ports`. (We can leave this in here, we just need to determine whether we need to support this additional field.)

shamsher31 replied:
Not sure why is this field added but you can refer more about it in this upstream PR https://github.com/istio/istio/pull/31253

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/deployment.yaml:
```diff
@@ -59,12 +68,14 @@
-  name: {{ $gateway.name | default "istio-ingressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
Once again, this will break deploying gateways to other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/gateways.yaml:
```diff
@@ -4,7 +4,7 @@ apiVersion: networking.istio.io/v1alpha3
 metadata:
   name: istio-autogenerated-k8s-ingress
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/istio-ingressgateway-route.yaml:
```diff
@@ -5,9 +5,9 @@ kind: Route
   name: {{ $gateway.name | default "istio-ingressgateway" }}
 {{- if or (not $gateway.namespace) (eq $gateway.name "istio-ingressgateway") }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/service.yaml:
```diff
@@ -1,9 +1,10 @@
-  name: {{ $gateway.name | default "istio-egressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/poddisruptionbudget.yaml:
```diff
@@ -3,8 +3,8 @@
-  name: {{ $gateway.name | default "istio-egressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/role.yaml:
```diff
@@ -2,14 +2,14 @@
-  name: {{ $gateway.name | default "istio-egressgateway" }}-sds
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}-sds
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/rolebindings.yaml:
```diff
@@ -2,17 +2,17 @@
-  name: {{ $gateway.name | default "istio-egressgateway" }}-sds
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}-sds
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/serviceaccount.yaml:
```diff
@@ -8,10 +8,10 @@ imagePullSecrets:
-  name: {{ $gateway.name | default "istio-egressgateway" }}-service-account
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}-service-account
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/autoscale.yaml:
```diff
@@ -3,8 +3,8 @@
-  name: {{ $gateway.name | default "istio-ingressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-egress/templates/autoscale.yaml:
```diff
@@ -3,8 +3,8 @@
-  name: {{ $gateway.name | default "istio-egressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/meshexpansion.yaml:
```diff
@@ -1,81 +0,0 @@
-{{ $gateway := index .Values "gateways" "istio-ingressgateway" }}
```
We need to determine if we need to modify anything in SMCP for this change.  We have some fields for mesh expansion, but they're not supported or tested, so maybe we just document them as being deprecated.

shamsher31 replied:
@luksa where can we do that?

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/networkpolicy.yaml:
```diff
@@ -4,9 +4,9 @@ kind: NetworkPolicy
+  namespace: {{ .Release.Namespace }}
 {{- else }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  namespace: {{ $gateway.namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/poddisruptionbudget.yaml:
```diff
@@ -3,8 +3,8 @@
-  name: {{ $gateway.name | default "istio-ingressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/preconfigured.yaml:
```diff
@@ -1,79 +0,0 @@
-{{ $gateway := index .Values "gateways" "istio-ingressgateway" }}
```
Similar to above, we need to determine if we need to deprecate any fields in SMCP.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/role.yaml:
```diff
@@ -2,8 +2,8 @@
-  name: {{ $gateway.name | default "istio-ingressgateway" }}-sds
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}-sds
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/rolebindings.yaml:
```diff
@@ -2,17 +2,17 @@
-  name: {{ $gateway.name | default "istio-ingressgateway" }}-sds
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}-sds
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/service.yaml:
```diff
@@ -3,8 +3,8 @@
-  name: {{ $gateway.name | default "istio-ingressgateway" }}
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/templates/serviceaccount.yaml:
```diff
@@ -8,8 +8,8 @@ imagePullSecrets:
-  name: {{ $gateway.name | default "istio-ingressgateway" }}-service-account
-  namespace: {{ $gateway.namespace | default .Release.Namespace }}
+  name: {{ $gateway.name }}-service-account
+  namespace: {{ .Release.Namespace }}
```
This needs to be reverted, or it will break installation of gateways in other namespaces.

rcernich commented on resources/helm/v2.1/gateways/istio-ingress/values.yaml:
```diff
@@ -24,7 +24,11 @@ gateways:
       name: https
       protocol: TCP
-      # This is the port where sni routing happens
+    - port: 15012
```
We should not be exposing this port over the gateway.  Our support for mesh expansion will automatically add this port.

shamsher31 replied:
Removed [here](https://github.com/maistra/istio-operator/pull/737/commits/7283c4555e644786340c06d3081bc919251bf5c9). Thanks.

rcernich commented on resources/helm/v2.1/global.yaml:
```diff
@@ -1,444 +0,0 @@
-# Global and common settings for installing Istio.
```
We may want to consider keeping this file, as it serves as global default values for all charts.  This will help maintain consistency.  That said, if upstream are manually updating individual values.yaml files for each chart, we run the risk of missing a change.  catch 22

shamsher31 replied:
Upstream already moved all the values from `global.yaml` to `values.yaml` and I think we already got those changes in rebase. I have also updated scripts to change values in `values.yaml` instead of `global.yaml`.

Upstream PR: https://github.com/istio/istio/issues/25184

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/files/gateway-injection-template.yaml:
```diff
@@ -0,0 +1,234 @@
+{{- $containers := list }}
```
We need to compare this with the sidecar injection template to pickup any changes we make there.

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/files/injection-template.yaml:
```diff
@@ -1,15 +1,45 @@
-  {{- if contains "/" .Values.global.proxy_init.image }}
-    image: "{{ .Values.global.proxy_init.image }}"
+  {{- if contains "/" (annotation .ObjectMeta `sidecar.istio.io/proxyImage` .Values.global.proxy_init.image) }}
+    image: "{{ annotation .ObjectMeta `sidecar.istio.io/proxyImage` .Values.global.proxy_init.image }}"
```
this change looks wrong.

dgn replied:
why? annotation func template looks like this:
```golang
func getAnnotation(meta metav1.ObjectMeta, name string, defaultValue interface{}) string
```

rcernich replied:
nm, i see that it's just allowing the init image to be overridden by an annotation.

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/files/injection-template.yaml:
```diff
@@ -117,8 +147,8 @@ template: |
-  {{- if contains "/" .Values.global.proxy_init.image }}
-    image: "{{ .Values.global.proxy_init.image }}"
+  {{- if contains "/" (annotation .ObjectMeta `sidecar.istio.io/proxyImage` .Values.global.proxy_init.image) }}
+    image: "{{ annotation .ObjectMeta `sidecar.istio.io/proxyImage` .Values.global.proxy_init.image }}"
```
this change looks wrong

dgn replied:
again, I think it's fine, see above. it just adds an indirection, it'll look at the annotation prior to the global value

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/files/injection-template.yaml:
```diff
@@ -445,27 +486,13 @@ template: |
-    traffic.sidecar.istio.io/excludeOutboundPorts: "{{ annotation .ObjectMeta `traffic.sidecar.istio.io/excludeOutboundPorts` .Values.global.proxy.excludeOutboundPorts }}"
-  {{- end }}
-    traffic.sidecar.istio.io/kubevirtInterfaces: "{{ index .ObjectMeta.Annotations `traffic.sidecar.istio.io/kubevirtInterfaces` }}"
-  {{- end }}
```
These look like they moved to the top of the file.  We need to make sure any changes we made are reflected there.  Specifically, we've had issues in the past with things like excluded/included ports/ips, etc.

shamsher31 replied:
These are moved directly to `annotations` from `podRedirectAnnot`. See [here](https://github.com/maistra/istio-operator/pull/737/files#diff-5da83be22dd99995fc488f94fad235a91932cddbcbf367c73e490a19b76fe4a5R10)

luksa replied:
Yup. `podRedirectAnnot` was removed. These annotations can now be added via the `annotations` config field.

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/templates/deployment.yaml:
```diff
@@ -94,10 +92,6 @@ spec:
             protocol: TCP
-          - containerPort: 15053
-            protocol: TCP
-          - containerPort: 8188
```
This is for discovery, although this change will need to be made in maistra/istio

shamsher31 replied:
This is back.

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/templates/mutatingwebhook.yaml:
```diff
@@ -9,26 +33,38 @@ metadata:
+        values:
+        - "true"
+{{- end }}
+{{- end }}
```
These changes don't look correct.

shamsher31 replied:
Sorry, can you please elaborate?

shamsher31 replied:
This is fixed.

dgn replied:
are you sure? it stil looks strange -- if `.Values.sidecarInjectorWebhook.useLegacySelectors` is enabled ,it will produce an invalid resource

dgn replied:
or maybe I'm misunderstanding how the mergeOverwrite above works

shamsher31 replied:
Added https://github.com/maistra/istio-operator/pull/756 to fix the issue.

rcernich commented on resources/helm/v2.1/istio-control/istio-discovery/values.yaml:
```diff
@@ -115,13 +94,41 @@ sidecarInjectorWebhook:
+  #         hello: world
+  #
+  # defaultTemplates: ["sidecar", "hello"]
+  defaultTemplates: []
```
We should consider disabling this functionality, as it will allow injection of arbitrary things.

shamsher31 replied:
Added disabling script [here](https://github.com/maistra/istio-operator/pull/737/commits/ddc56f70607b49bf921167d8ef7032309f15e7cd). Thanks.

rcernich commented on resources/helm/v2.1/mesh-config/templates/telemetryv2_1.6.yaml:
```diff
@@ -1,612 +0,0 @@
-{{- if and .Values.telemetry.enabled .Values.telemetry.v2.enabled }}
```
We need to keep this, as maistra 2.0 is based on istio 1.6

shamsher31 replied:
@rcernich 1.6 and 1.7 telemetry files are removed from upstream. Do you want me to copy both into the overlay folder to make sure we still have support for the older version?

rcernich replied:
We only need 1.6, as we never had a release with 1.7.

shamsher31 replied:
Perfect. I have added back 1.6 telemetry https://github.com/maistra/istio-operator/pull/737/commits/fed344593f14912727970121c194e91126cb2ba7

Commit:
keep the default values for namespace

Commit:
disable defaultTemplates

shamsher31 commented on resources/helm/v2.1/istio-control/istio-discovery/templates/role.yaml:
```diff
@@ -21,11 +21,3 @@ rules:
-
-# For MeshFederation support
-- apiGroups: ["maistra.io"]
-  resources: ["meshfederations", "meshfederations/status", "federationstatuses", "federationstatuses/status", "serviceexports", "serviceexports/status", "serviceimports", "serviceimports/status"]
```
@dgn @rcernich I think we should keep this in a separate file and merge it in `role.yaml` during `make gen`?

shamsher31 replied:
Added mesh federation role as an overly [here](https://github.com/maistra/istio-operator/pull/737/commits/2e2717af4c9ba8925ad755dc9666641d38ed3f33).

dgn replied:
why did this disappear? did we change the approach during the rebase?

rcernich replied:
I believe all the changes for mesh federation have been added to the charts in maistra/istio.  we should not be adding them to the overlays.

shamsher31 replied:
Right, I have removed the overlay because now after the latest rebase, all the mesh federation changes are coming from maistra/istio.

Commit:
remove special perm

Commit:
do not expose port 15012

Commit:
fix wrong indentation and add mtls to mesh-config values

Commit:
add missing end for mutatingwebhook.yaml

shamsher31 commented on pkg/controller/versions/strategy_v2_1.go:
```diff
@@ -312,12 +311,8 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
 	}
 
-	// Read in global.yaml
-	values, err := chartutil.ReadValuesFile(path.Join(v.GetChartsDir(), "global.yaml"))
```
@luksa if we want to bring back `global.yaml`, we will have to revert this code.

shamsher31 commented on build/patch-charts.sh:
```diff
@@ -368,15 +386,12 @@ function patchSidecarInjector() {
-    includeInboundPorts: "*"' ${HELM_DIR}/global.yaml
+    includeInboundPorts: "*"' ${HELM_DIR}/istio-control/istio-discovery/values.yaml
   # status port is incorrect
-  sed_wrap -i -e 's/statusPort: 15020$/statusPort: 15021/' ${HELM_DIR}/global.yaml
```
@luksa Also this needs to be reverted for `global.yaml` file.

Commit:
add tracer section to gateway values.yaml

Commit:
set fsGroup to ProxyGID instead of ProxyUID

luksa commented:
/test istio-operator_unittests

shamsher31 commented on resources/helm/overlays/istio-control/istio-discovery/templates/meshclusterrole.yaml:
```diff
@@ -11,3 +11,18 @@ rules:
+  - apiGroups:
+    - networking.k8s.io
+    resources:
+    - ingressclasses
```
Looks like this was added in 1.9 to replace  `kubernetes.io/ingress.class` annotation.
https://istio.io/latest/docs/tasks/traffic-management/ingress/kubernetes-ingress/#specifying-ingressclass

rcernich replied:
jfyi, we should be disabling the ingress controller in istio.

shamsher31 replied:
@rcernich Is this https://github.com/maistra/istio-operator/pull/737/commits/9e7b819182c5555a37f8511a6ad9520970bfaec8 the correct way to disable the Ingress controller?

shamsher31 replied:
This is good for now. https://github.com/maistra/istio-operator/pull/737#commitcomment-52972441

Commit:
fix test

Commit:
provide istio version

Commit:
add back global.yaml

Commit:
fix test

Commit:
make gen

dgn commented on build/patch-charts.sh:
```diff
@@ -58,6 +58,11 @@ function patchTemplates() {
   mv ${HELM_DIR}/base/templates/role.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/role.yaml
   mv ${HELM_DIR}/base/templates/rolebinding.yaml ${HELM_DIR}/istio-control/istio-discovery/templates/rolebinding.yaml
 
+  # add internal annotation to Mesh Fedration Role.
```
nit: typo

brian-avery replied:
```suggestion
  # add internal annotation to Mesh Federation Role.
```

shamsher31 replied:
Thanks

dgn commented on resources/helm/overlays/istio-control/istio-discovery/templates/meshfederationrole.yaml:
```diff
@@ -0,0 +1,16 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
should we include this file in maistra/istio? how did this work previously?

rcernich replied:
isn't this coming from maistra/istio?

dgn replied:
it's an overlay, so I guess no?

rcernich replied:
yeah, as mentioned in my general comment, all of the federation settings should be coming from maistra/istio

shamsher31 replied:
Right, I have removed the overlay because now after the latest rebase, all the mesh federation changes are coming from maistra/istio.

luksa commented on resources/helm/overlays/istio-control/istio-discovery/templates/meshclusterrole.yaml:
```diff
@@ -11,3 +11,18 @@ rules:
+    verbs:
+    - list
+    - watch
+  - apiGroups:
+    - ""
+    resources:
+    - namespaces
+    verbs:
+    - list
+    - watch
```
This is no longer required.

shamsher31 replied:
Removed.

luksa commented on pkg/controller/versions/strategy_v2_1.go:
```diff
@@ -312,12 +311,8 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
-		return nil, fmt.Errorf("error reading global.yaml file")
-	}
-	values.MergeInto(spec.Istio.GetContent())
+	// global.yaml is removed and combined into values.yaml of each chart
```
Didn't you add global.yaml back? Doesn't this code now need to be here again?

shamsher31 replied:
Yes, I added it back because chart rendering was failing due to a lot of missing global values and it was hard to figure out which changes are from upstream and what is coming from maistra charts.

Thanks, I will revert it back.

dgn commented on resources/helm/v2.1/istio-control/istio-discovery/files/injection-template.yaml:
```diff
@@ -81,7 +111,7 @@ template: |
     {{- end }}
   {{- else }}
     {{- if .Values.global.proxy.resources }}
-      {{ toYaml .Values.global.proxy.resources | indent 4 }}
```
interesting. this change suggests that proxy resource config didn't work previously?

dgn commented on resources/helm/v2.1/istio-control/istio-discovery/templates/deployment.yaml:
```diff
@@ -168,29 +160,9 @@ spec:
-          - name: CENTRAL_ISTIOD
-            value: "{{ $.Values.global.centralIstiod | default "false" }}"
-{{- if .Values.global.tls }}
-{{- if .Values.global.tls.minProtocolVersion }}
```
hmm.. this stuff is part of the TLS customization changeset we have in maistra. it shouldn't get lost

shamsher31 replied:
It's back now after the latest rebase.

dgn commented on resources/helm/v2.1/istio-control/istio-discovery/templates/service.yaml:
```diff
@@ -24,9 +24,6 @@ spec:
     - port: 15014
       name: http-monitoring # prometheus stats
       protocol: TCP
-    - port: 8188
```
we stil need this afaik. it's for federation

shamsher31 replied:
After latest rebase Federation port settings are back.

dgn reviewed (commented):
it looks like some chart changes related to federation disappeared during the maistra/istio rebase. also, the TLS cipher options are gone

rcernich commented on resources/helm/overlays/istio-telemetry/prometheus/values.yaml:
```diff
@@ -53,4 +53,16 @@ prometheus:
   # When this option is set as false, this certificate provisioning mechanism is disabled.
   provisionPrometheusCert: true
 
+global:
```
are these necessary if global.yaml is configuring them?

shamsher31 replied:
We previously removed merging `global.yaml` file logic, but now we added it back so it's safe to remove it from promotes values.

rcernich reviewed (commented):
I think we'll need to update the charts, once all the changes are in the maistra/istio 1.9 branch.  There appear to be a lot of changes related to federation that are missing here.

Also, we need to be careful about global.yaml.  I see a number of changes related to various values.yaml files, and anything in global.yaml will override those, so we need to ensure that global.yaml is correct.

Commit:
update to latest charts including meshfederation

shamsher31 commented:
@dgn @rcernich @luksa I rebased with the latest merge of Marko's PR for 1.9 changes and I can see a lot of mesh federation work is now updated in charts.

2 days ago, Marko's PR was not merged in the `maistra-2.1-istio-1.9` branch hence most of the latest changes like mesh federation, service import, export, and TLS settings were missing.

Commit:
fix test

Commit:
remove prom vlaues as its handled by global values

Commit:
add subjectaccessreviews perm

shamsher31 commented:
/test istio-operator_unittests

Commit:
disable ingress controller

Commit:
bring back 1.6 telemetry to avoid upgrade failure

luksa reviewed (approved):
LGTM, but let's review the outstanding comments and submit fixes in follow-up PRs.
